src/soroban/server.ts

```diff
@@ -235,6 +235,93 @@ export class Server {
+   * deployed on the Soroban network. The WASM bytecode represents the executable
+   * code of the contract.
+   *
+   * @param {string} contractId    the contract ID containing the
+   *    WASM bytecode to retrieve
+   *
+   * @returns {Promise<Buffer>}   a Buffer containing the WASM bytecode
+   *
+   * @throws {Error} If the contract or its associated WASM bytecode cannot be
+   * :
+   *    found on the network.
+   *
+   * @example
+   * const contractId = "CCJZ5DGASBWQXR5MPFCJXMBI333XE5U3FSJTNQU7RIKE3P5GN2K2WYD5";
+   * server.getContractWasmByContractId(contractId).then(wasmBuffer => {
+   *   console.log("WASM bytecode length:", wasmBuffer.length);
+   *   // ... do something with the WASM bytecode ...
+   * }).catch(err => {
+   *   console.error("Error fetching WASM bytecode:", err);
+   * });
+   */
+  public async getContractWasmByContractId(
+    contractId: string
+  ): Promise<Buffer> {
+    const contractLedgerKey = new Contract(contractId).getFootprint();
+    const response = await this.getLedgerEntries(contractLedgerKey);
+    if (!response.entries.length || !response.entries[0]?.val) {
+      return Promise.reject({code: 404, message: `Could not obtain contract hash from server`});
+    }
+
+    const wasmHash = response.entries[0].val
+      .contractData()
+      .val()
+      .instance()
+      .executable()
+      .wasmHash();
+
+    return this.getContractWasmByHash(wasmHash);
+  }
+
+  /**
+   * Retrieves the WASM bytecode for a given contract hash.
+   *
+   * This method allows you to fetch the WASM bytecode associated with a contract
+   * deployed on the Soroban network using the contract's WASM hash. The WASM bytecode
+   * represents the executable code of the contract.
+   *
+   * @param {Buffer} wasmHash    the WASM hash of the contract
+   *
+   * @returns {Promise<Buffer>}   a Buffer containing the WASM bytecode
+   *
+   * @throws {Error} If the contract or its associated WASM bytecode cannot be
+   *    found on the network.
+   *
+   * @example
+   * const wasmHash = Buffer.from("...");
+   * server.getContractWasmByHash(wasmHash).then(wasmBuffer => {
+   *   console.log("WASM bytecode length:", wasmBuffer.length);
+   *   // ... do something with the WASM bytecode ...
+   * }).catch(err => {
+   *   console.error("Error fetching WASM bytecode:", err);
+   * });
+   */
+  public async getContractWasmByHash(
+    wasmHash: Buffer
+  ): Promise<Buffer> {
+    const ledgerKeyWasmHash = xdr.LedgerKey.contractCode(
+      new xdr.LedgerKeyContractCode({
+        hash: wasmHash
+      })
+    );
+
+    const responseWasm = await this.getLedgerEntries(ledgerKeyWasmHash);
+    if(!responseWasm.entries.length || !responseWasm.entries[0]?.val){
+      return Promise.reject({code: 404, message: `Could not obtain contract wasm from server`});
+    }
+    const wasmBuffer = responseWasm.entries[0].val.contractCode().code();
+
+    return Buffer.from(wasmBuffer);
```

Comment:
On that note, why encode as a Buffer here? The less we pass around Buffers, the easier it will be to remove that dependency someday. I think a "JIT conversion to Buffer" strategy is probably the best call, rather than eagerly converting to Buffer and sprinkling Buffer arguments throughout our codebase.

Reply:
Good catch, no need there.